sklearn/impute/_base.py

```diff
@@ -270,7 +280,10 @@ class SimpleImputer(_BaseImputer):
 
     _parameter_constraints: dict = {
         **_BaseImputer._parameter_constraints,
-        "strategy": [StrOptions({"mean", "median", "most_frequent", "constant"})],
+        "strategy": [
+            StrOptions({"mean", "median", "most_frequent", "constant"}),
+            type(np.min),
```

Comment:
To look for precedent:
```
$ git grep '".*or .allable'
sklearn/feature_extraction/text.py:        if self.analyzer != "word" or callable(self.analyzer):
sklearn/feature_selection/_base.py:    getter : "auto", str or callable
sklearn/feature_selection/tests/test_from_model.py:    """Check max_features_ and output shape for callable max_features."""
sklearn/gaussian_process/kernels.py:              "rbf", "laplacian", "sigmoid", "cosine"} or callable, \
sklearn/neighbors/_nearest_centroid.py:                    "manhattan and for callables was deprecated in version "
sklearn/preprocessing/_encoders.py:    feature_name_combiner : "concat" or callable, default="concat"
```

Reply:
Looking at the docstring of `validate_parameter_constraints` you want to use
```suggestion
            callable,
```
rather than `typing.Callable`.
However, it might make sense to extend `validate_parameter_constraints` to support any object that hasattr `__instancecheck__` as a type.